### PR TITLE
Problem1

### PR DESCRIPTION
**Pull request by Tohanean**
Given an integer array of size N, compute the sum of all even numbers in this array

### CODE REVIEW

File: EvenSum.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class EvenSum {
+    fun evenSum(array: Array<Int>) : Int
+    {
+        var i:Int = 0
+        var sum:Int = 0
+        for(i in 0..array.size-1 step 1)
+        {
+            println(array[i])
+        }
```
Comment (alexandru-calinoiu) [inline EvenSum.kt:9]: Not sure why printing helps here, it was not in the req of the problem.

If one does want to keep printing please consider merging the 2 loops.

File: EvenSum.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    fun evenSum(array: Array<Int>) : Int
+    {
+        var i:Int = 0
+        var sum:Int = 0
+        for(i in 0..array.size-1 step 1)
+        {
+            println(array[i])
+        }
+
+        for(i in 0..array.size-1 step 1)
```
Comment (alexandru-calinoiu) [inline EvenSum.kt:11]: Most modern languages have nicer ways of iterating over arrays, such is the case with kotlin, can you look into an alternative for this for index ?

File: EvenSumTest.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import com.sun.tools.javac.Main
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+internal class MainKtTest {
```
Comment (alexandru-calinoiu) [inline EvenSumTest.kt:5]: Really cool that you wrote tests 👍🏽

File: EvenSumTest.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        val num = arrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+        assertEquals(expected, testSample1.evenSum(num))
+    }
+
+    @Test
+    fun testSum2() {
+        val expected = 76
+        val num = arrayOf(2,4,3,5,20,1,8,19,42,13)
+        assertEquals(expected, testSample1.evenSum(num))
+    }
```
Comment (alexandru-calinoiu) [inline EvenSumTest.kt:21]: This tests are pretty much the same, I can't imagine how one will fail an the other will not, I would suggest to have tests that cover more of the edge cases, like empty array, an array with all number odd ....

File: Main.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+fun main() {
```
Comment (alexandru-calinoiu) [inline Main.kt:1]: Main is not required, but I can see why you created it, proving the solution via tests is the best way.

**Review by alexandru-calinoiu — COMMENTED** [review]
Given that this is most likely your first interaction with kotlin it did worked out decent, left some comments.

One general aspect is that I suggest you use the build-in code formater from your editor.

File: EvenSum.kt (hunk 1 above)
Comment (Tohanean) [reply EvenSum.kt:9]: You are right, did it for trying things, and  I should've removed it as it was not required.

File: EvenSumTest.kt (hunk 4 above)
Comment (Tohanean) [reply EvenSumTest.kt:21]: Thank you for your input, now I see more clearly why one would use them.

File: EvenSum.kt (hunk 2 above)
Comment (Tohanean) [reply EvenSum.kt:11]: One solution i found was this:   for(i in array.indices) { if(array[i]%2 == 0) sum+= array[i]}
